lib/helpers/spawn.js

```diff
@@ -44,11 +44,16 @@ module.exports.sync = function (cmd, params, cwd, onFail) {
 
   if (runner.status || runner.error) {
     warn()
-    warn(`⚠️  Command "${cmd}" failed with exit code: ${runner.status}`)
+    if( cmd === 'firebase' && runner.status === 1) {
+      log(`⚠️  Command "${cmd}" exited, using defaults`)
+    } else{
+      warn(`⚠️  Command "${cmd}" failed with exit code: ${runner.status}`)
+    }
+
     if (runner.status === null) {
       warn(`⚠️  Please globally install "${cmd}"`)
     }
-    onFail && onFail()
-    process.exit(1)
+    if (onFail && onFail(runner)) process.exit(1)
```

Comment:
This may need to be changed back or refactored to be more concise. This was for getting the result of spawn in order to handle the error in the mode helper